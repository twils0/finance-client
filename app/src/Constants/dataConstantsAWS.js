// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
