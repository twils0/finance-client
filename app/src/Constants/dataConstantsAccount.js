export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
