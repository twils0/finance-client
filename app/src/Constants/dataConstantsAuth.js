export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
