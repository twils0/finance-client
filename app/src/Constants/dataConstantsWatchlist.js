export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
