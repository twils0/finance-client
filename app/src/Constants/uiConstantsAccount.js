export const actionTypes = {
  SET_CURRENT_FORM: 'ACCOUNT_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'ACCOUNT_SET_FORM_HEIGHT',
  SET_FORM_EDIT: 'ACCOUNT_SET_FORM_EDIT',
  SET_INPUT_VALUE_ERROR: 'ACCOUNT_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'ACCOUNT_SET_BUTTON_TEXT',
  SET_BUTTON_VISIBLE: 'ACCOUNT_SET_BUTTON_VISIBLE',
};

export const formNames = {
  CODE: 'codeForm',
  PROFILE: 'profileForm',
  BILLING: 'billingForm',
  CHANGE_PASSWORD: 'changePasswordForm',
  DELETE_ACCOUNT: 'deleteAccountForm',
};

export const buttonNames = {
  PROFILE: 'profileButton',
  BILLING: 'billingButton',
  CHANGE_PASSWORD: 'changePasswordButton',
  DELETE_ACCOUNT: 'deleteAccountButton',
  FIRST: 'firstButton',
  SECOND: 'secondButton',
};

export const buttonTexts = {
  RESEND: 'Resend',
  SUBMIT: 'Submit',
  EDIT: 'Edit',
  UPDATE: 'Update',
  CANCEL: 'Cancel',
  DELETE: 'Delete',
  CONFIRM: 'Confirm',
};

export const inputNames = {
  [formNames.CODE]: {
    CODE_PHONE: 'codeCodePhone',
  },
  [formNames.PROFILE]: {
    NAME: 'profileName',
    EMAIL: 'profileEmail',
    EMAIL_ADDITIONAL: 'profileEmailAdditional',
    PHONE: 'profilePhone',
  },
  [formNames.BILLING]: {
    NAME_ON_CARD: 'billingNameOnCard',
    PROMO_CODE: 'billingPromoCode',
    STRIPE: 'billingStripe',
  },
  [formNames.CHANGE_PASSWORD]: {
    OLD_PASSWORD: 'changePasswordOldPassword',
    NEW_PASSWORD: 'changePasswordNewPassword',
    NEW_PASSWORD2: 'changePasswordNewPassword2',
  },
  [formNames.DELETE_ACCOUNT]: {
    CONFIRM: 'deleteAccountConfirm',
  },
};
