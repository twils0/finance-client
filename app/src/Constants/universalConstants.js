import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
