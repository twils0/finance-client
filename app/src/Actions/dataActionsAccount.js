import { actionTypes } from '../Constants/dataConstantsAccount';

export const setAccountStatus = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'id')) {
    throw new Error(`Please enter a value for the 'id' key - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'status')) {
    throw new Error(`Please enter a value for the 'status' key - ${JSON.stringify(payload)}`);
  }

  return {
    type: actionTypes.SET_ACCOUNT_STATUS,
    payload,
  };
};

export const setField = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'id')) {
    throw new Error(`Please enter a value for the 'id' key - ${JSON.stringify(payload)}`);
  }
  if (!Object.prototype.hasOwnProperty.call(payload, 'value')) {
    throw new Error(`Please enter a value for the 'value' key - ${JSON.stringify(payload)}`);
  }

  return {
    type: actionTypes.SET_FIELD,
    payload,
  };
};

export const setFields = (payload) => {
  Object.keys(payload).forEach((key) => {
    const field = payload[key];

    if (!Object.prototype.hasOwnProperty.call(field, 'id')) {
      throw new Error(`Please enter a value for the 'id' key - ${JSON.stringify(field)}`);
    }
    if (!Object.prototype.hasOwnProperty.call(field, 'value')) {
      throw new Error(`Please enter a value for the 'value' key - ${JSON.stringify(field)}`);
    }
  });

  return {
    type: actionTypes.SET_FIELDS,
    payload,
  };
};

export const resetAccountState = () => ({ type: actionTypes.RESET_ACCOUNT_STATE });
