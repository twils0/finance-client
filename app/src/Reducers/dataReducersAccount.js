import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames, fieldNames } from '../Constants/dataConstantsAccount';

export const initialState = {
  status: {
    [statusNames.AWS_FIELDS]: {
      id: statusNames.AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.STRIPE_FIELDS]: {
      id: statusNames.STRIPE_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_TOKEN]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_TOKEN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_REQUEST]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_REQUEST,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_DB_FIELDS]: {
      id: statusNames.UPDATE_DB_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_AWS_FIELDS]: {
      id: statusNames.UPDATE_AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
  },
  fields: {
    list: [...Object.values(fieldNames)],
    [fieldNames.NAME]: {
      id: fieldNames.NAME,
      value: '',
    },
    [fieldNames.EMAIL]: {
      id: fieldNames.EMAIL,
      value: '',
    },
    [fieldNames.EMAIL_ADDITIONAL]: {
      id: fieldNames.EMAIL_ADDITIONAL,
      value: '',
    },
    [fieldNames.PHONE]: {
      id: fieldNames.PHONE,
      value: '',
    },
    [fieldNames.NAME_ON_CARD]: {
      id: fieldNames.NAME_ON_CARD,
      value: '',
    },
    [fieldNames.PROMO_CODE]: {
      id: fieldNames.PROMO_CODE,
      value: '',
    },
    [fieldNames.PROMO_CODE_VALID]: {
      id: fieldNames.PROMO_CODE_VALID,
      value: true,
    },
    [fieldNames.CUSTOMER]: {
      id: fieldNames.CUSTOMER,
      value: '',
    },
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_ACCOUNT_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const fields = (state = initialState.fields, action) => {
  switch (action.type) {
    case actionTypes.SET_FIELD:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_FIELDS:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.fields;
    default:
      return state;
  }
};

const dataReducersAccount = combineReducers({
  status,
  fields,
});

export default dataReducersAccount;
