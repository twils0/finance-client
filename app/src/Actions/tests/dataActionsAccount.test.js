import configureMockStore from 'redux-mock-store';
import { initialState } from '../../Reducers/dataReducersAccount';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { actionTypes } from '../../Constants/dataConstantsAccount';
import { setAccountStatus, setField, setFields, resetAccountState } from '../dataActionsAccount';

const middleware = [];
const mockStore = configureMockStore(middleware);

describe('dataActionsAccount', () => {
  describe('setAccountStatus', () => {
    it("fails when not provided a payload with an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { status: requestStatusTypes.SUCCESS };

      expect(() => store.dispatch(setAccountStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'status' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setAccountStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', status: requestStatusTypes.SUCCESS };
      const expectedActions = [
        {
          type: actionTypes.SET_ACCOUNT_STATUS,
          payload,
        },
      ];

      store.dispatch(setAccountStatus(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setField', () => {
    it("fails when not provided a payload with an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { value: 'testValue' };

      expect(() => store.dispatch(setField(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'value' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setField(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', value: 'testValue' };
      const expectedActions = [
        {
          type: actionTypes.SET_FIELD,
          payload,
        },
      ];

      store.dispatch(setField(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setFields', () => {
    it("fails when not provided a payload with an 'id' key", () => {
      const store = mockStore(initialState);

      const payload = {
        testId1: { id: 'testId1', value: 'testValue1' },
        testId2: { value: 'testValue2' },
      };

      expect(() => store.dispatch(setFields(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'value' key", () => {
      const store = mockStore(initialState);

      const payload = {
        testId1: { id: 'testId1', value: 'testValue1' },
        testId2: { id: 'testId2' },
      };

      expect(() => store.dispatch(setFields(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = {
        testId1: { id: 'testId1', value: 'testValue1' },
        testId2: { id: 'testId2', value: 'testValue2' },
      };
      const expectedActions = [
        {
          type: actionTypes.SET_FIELDS,
          payload,
        },
      ];

      store.dispatch(setFields(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('resetAccountState', () => {
    it('creates the correct action', () => {
      const store = mockStore(initialState);

      const expectedActions = [
        {
          type: actionTypes.RESET_ACCOUNT_STATE,
        },
      ];

      store.dispatch(resetAccountState());

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });
});
