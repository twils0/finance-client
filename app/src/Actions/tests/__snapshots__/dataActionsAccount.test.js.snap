// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataActionsAccount setAccountStatus fails when not provided a payload with a 'status' key 1`] = `"Please enter a value for the 'status' key - {\\"id\\":\\"testId\\"}"`;

exports[`dataActionsAccount setAccountStatus fails when not provided a payload with an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"status\\":\\"SUCCESS\\"}"`;

exports[`dataActionsAccount setField fails when not provided a payload with a 'value' key 1`] = `"Please enter a value for the 'value' key - {\\"id\\":\\"testId\\"}"`;

exports[`dataActionsAccount setField fails when not provided a payload with an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"value\\":\\"testValue\\"}"`;

exports[`dataActionsAccount setFields fails when not provided a payload with a 'value' key 1`] = `"Please enter a value for the 'value' key - {\\"id\\":\\"testId2\\"}"`;

exports[`dataActionsAccount setFields fails when not provided a payload with an 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"value\\":\\"testValue2\\"}"`;
